extra/modules/fiftyone-devicedetection/src/main/java/org/prebid/server/hooks/modules/fiftyone/devicedetection/v1/hooks/FiftyOneDeviceDetectionRawAuctionRequestHook.java

```diff
@@ -0,0 +1,157 @@
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.model.ModuleContext;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.model.InvocationResultImpl;
+import org.prebid.server.hooks.v1.InvocationAction;
+import org.prebid.server.hooks.v1.InvocationResult;
+import org.prebid.server.hooks.v1.InvocationStatus;
+import org.prebid.server.hooks.v1.auction.AuctionInvocationContext;
+import org.prebid.server.hooks.v1.auction.AuctionRequestPayload;
+import org.prebid.server.hooks.v1.auction.RawAuctionRequestHook;
+import io.vertx.core.Future;
+import org.prebid.server.settings.model.Account;
+import org.prebid.server.util.ObjectUtil;
+
+import jakarta.annotation.Nonnull;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+public class FiftyOneDeviceDetectionRawAuctionRequestHook implements RawAuctionRequestHook {
+    private static final String CODE = "fiftyone-devicedetection-raw-auction-request-hook";
+
+    private final AccountFilter accountFilter;
+    private final DeviceEnricher deviceEnricher;
+
+    public FiftyOneDeviceDetectionRawAuctionRequestHook(@Nonnull AccountFilter accountFilter,
+                                                        @Nonnull DeviceEnricher deviceEnricher) {
+        this.accountFilter = Objects.requireNonNull(accountFilter);
+        this.deviceEnricher = Objects.requireNonNull(deviceEnricher);
+    }
+
+    @Override
+    public String code() {
+        return CODE;
+    }
+
+    @Override
+    public Future<InvocationResult<AuctionRequestPayload>> call(AuctionRequestPayload payload,
+                                                                AuctionInvocationContext invocationContext) {
+        final ModuleContext oldModuleContext = (ModuleContext) ObjectUtil.getIfNotNull(
+                invocationContext,
+                AuctionInvocationContext::moduleContext);
+
+        if (!isAccountAllowed(invocationContext)) {
+            return Future.succeededFuture(
+                    InvocationResultImpl.<AuctionRequestPayload>builder()
+                            .status(InvocationStatus.success)
+                            .action(InvocationAction.no_action)
+                            .moduleContext(oldModuleContext)
+                            .build());
+        }
+
+        final ModuleContext moduleContext = addEvidenceToContext(
+                oldModuleContext,
+                builder -> collectEvidence(builder, payload.bidRequest()));
+
+        return Future.succeededFuture(
+                InvocationResultImpl.<AuctionRequestPayload>builder()
+                        .status(InvocationStatus.success)
+                        .action(InvocationAction.update)
+                        .payloadUpdate(freshPayload -> updatePayload(freshPayload, moduleContext.collectedEvidence()))
```

Comment:
There can be cases when nothing happens (no updates for device, etc), so `InvocationAction` must be `no_action` in this case, but would be `update`.

If it's not critical  - nvm then.

Reply:
:heavy_check_mark: Added a check for blank `device.ext.fiftyonedegrees_deviceId` as update precondition.
:heavy_check_mark: Added unit test.